test/internal/request/complement.spec.js

```diff
@@ -0,0 +1,113 @@
+var expect = require("chai").expect;
+var complement = require("../../../lib/request/complement");
+var findPartialIntersections = require("../../../lib/request/complement").findPartialIntersections;
+
+describe("complement", function() {
+    it("returns null if no paths are provided", function() {
+        expect(complement([], [], {})).to.be.null;
```

Comment:
Ending an assertion chain with a property accessor is dangerous because the following test would also pass:

```js
        expect(complement([], [], {})).to.be.nul;
```

I've been bitten by this in the past. To solve it, I recommend every assertion chain ending with a method call:

```suggestion
        expect(complement([], [], {})).to.equal(null);
```

Reply:
Wasn't aware of that, thanks.